examples/bazel_simple/WORKSPACE

```diff
@@ -0,0 +1,53 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
+    name = "com_google_protobuf",
+    commit = "09745575a923640154bcf307fba8aedff47f240a",
```

Comment:
It seems that protobuf-python is being installed from `pip`, and presumably the grpcio-tools version of protoc could be installed similarly. What is the git checkout of the protobuf repo being used for?

Reply:
This is more of a matter of personal preference. So for example using this you can make use of a specific version of protoc, and that selection will be entirely deterministic and hermetic. The pip rules are not considered hermetic. There is a bit of a divided group of people at bazel on how deterministic tool-chains should be (i.e. should you use system installations? or use a determinsitically built toolchain). 

You are right in that, this is not necessarily 'required', my personal preference is that all toolchains should be as deterministic as possible, at the expense of downloads/compile time. 

For context [here](https://github.com/grailbio/bazel-toolchain) is a completely deterministic c++ toolchain for bazel. Which fetches the entire toolchain and the sysroot, and matches them against checksums.

I could potentially make this optional?

Reply:
The protobuf building & installation is quite heavy, so it being a mandatory dependency might be quite annoying.

The grpcio-tools version of protoc is Python-based AFAIK, so even if one would go the git route instead of pip install, it might be faster and easier to get working.

I think bazel has some kind of mechanism of providing rules from the upper level project, so might there be some kind of rule for "get me working protoc", that the user could then override to be even empty if protoc is already available from `apt-get` or similar.

Reply:
Chiming in from the make-it-hermetic camp, this is idiomatic for Bazel (and for how it's used in Bazel's parent org). As examples, when you use Bazel to build Go code it downloads the entire Go SDK (so Bazel has its own repeatable copy rather than using the system copy), and when you run unit tests it downloads what looks to be the entire JDK. Heavy-weight, in this case, is in the eye of the beholder. :-) 

If you're wondering whether the typical Bazel user would be surprised to see it downloading all of these repositories just to enable a nanopb_proto_library rule or to build the generator, no, they would not be surprised. It would be considered totally normal.